core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -3639,6 +3642,37 @@ public void checkAllPredicatesAndTableSetOp(String sql) {
         is(mq.getPopulationSize(rel, bitSetOf(0))));
   }
 
+  /**
+   * Test that RelMdPopulationSize is calculated based on the RelMetadataQuery#getRowCount().
+   *
+   * @see <a href="https://issues.apache.org/jira/browse/CALCITE-5647">[CALCITE-5647]</a>
+   */
+  @Test public void testPopulationSizeFromValues() {
```

Comment:
This isn't really displaying a bug that is being fixed.
From the JIRA comments I understand that it's difficult to create a small reproduction.

Reply:
The bug is that the population size was being calculated without utilizing RelMetadataQuery#getRowCount(). The idea was that this test demonstrates the getRowCount() call is being used.

Does that clarify the bug and purpose of the test? This sort of test you'd probably want to spy() instead but I didn't want to introduce Mockito into the core sub-module just for this.

Reply:
Yes, the test is still artificial.